### ISSUE TITLE
Container syncing between release and nightly is not workin

### ISSUE THREAD

**Issue by gabrielambuehl** [mozilla/multi-account-containers#1802]
### Actual behavior

Setup containers addon to sync on release firefox and nightly on two machines. Release happily syncs containers back and forth, nightly never gets the containers. Regular history sync etc works across all instances.

### Expected behavior

Containers sync between release and nightly instances.

### Notes
Windows 10 on both x64 and aarch64.

**Comment by MorgothSauron** [comment]
I'm unable to sync anything between Linux, macOS and Windows using the latest stable version on each OS.

**Comment by gabrielambuehl** [comment]
Interesting - I setup one of my Windows machines today because Windows Update was hosed and now stable also won't sync with the other one either...

**Comment by woodpecs** [comment]
I was also experiencing this issue between a Windows and Linux machine with nightly and stable. Tried some troubleshooting and turns out add on sync in Firefox sync settings needs to be enabled. Do you guys have that enabled? I'm not sure if that is the designed behavior, but I guess it makes sense. Ideally I would like to not sync my add-ons between the two because Linux has other specific ones I don't want on Windows. I was thinking the containers would sync with your Firefox account specifically but it seems add-on syncing also must be enabled, since Firefox containers is technically an add-on? 

If that is the case, then just now I got my Linux machine with Firefox stable and my Windows machine with Firefox nightly to sync containers.

**Comment by MorgothSauron** [comment]
> Do you guys have that enabled?

Yes, I have Firefox sync enabled. I have it enabled at the browser level and I also setup sync when I install the extension. I still don't understand why I have to setup sync in the addon when it's already enabled in the browser. 

To be honest I think the sync for this addon just broken. When I removed it from one browser, it wasn't removed on the other browsers. Everything syncs fine, except this addon. I hope the sync problem(s) will be solved because this extension has a lot of potential.

**Comment by mcepl** [comment]
I just had to gave up on containers completely, because with Linux Nightly I get too many broken profiles, and containers never synced for me, not even once. Oh well.

**Comment by rpl** [comment]
@maxxcrawford the about:sync extension (https://addons.mozilla.org/en-US/firefox/addon/about-sync/) may be useful to investigate this issue.

Some other notes that may be related and useful to get to the bottom of this issue:
- storage.sync implementation and server side backend did change in Firefox 79 (https://blog.mozilla.org/addons/2020/07/09/changes-to-storage-sync-in-firefox-79/), which has been release around the end of July (and the time when this issue has been filed does match that):
  - the new and old implementation can't sync with each other and so Firefox instances with versions < 79 can't sync with Firefox versions with versions >= 79
  - the new implementation does enforce a client side quota, and an error will be throw if the extension tries to add more data while being over quota (the extension will still be able to remove data and can check how much storage is being used using `browser.storage.sync.getBytesInUse()`
- There was an issue we fixed recently that may also prevent storage.sync data from syncing successfully if one of the extension installed did call storage.sync.clear ([Bug 1667985 - webext-storage incoming tombstones not handled correctly](https://bugzilla.mozilla.org/show_bug.cgi?id=1667985), fixed in Firefox 84 by [Bug 1675190](https://bugzilla.mozilla.org/show_bug.cgi?id=1675190))

**Comment by mcepl** [comment]
@rpl not my problem: a) I have never successfully recovered containers on new profile, b) moving just between FF 82 and FF 85 (so still well over 79).

**Comment by rpl** [comment]
> @rpl not my problem: a) I have never successfully recovered containers on new profile, b) moving just between FF 82 and FF 85 (so still well over 79).

the part related to the incompatibility between the old and new storage.sync backends is more likely the issue behind the original report, the more recent ones are more likely to be unrelated to that. 

Nevertheless, the more recent report may be related to the other issues describe in my previous comment (e.g. related to the client-side quota limits if the extension is not handling that scenario, or to Bug 1667985).

@mcepl If you are still able to reproduce the issue, you may try to install the about:sync addon, opening about:async in a new tab and then force a sync using "Sync Now" action in the Firefox Account toolbar button popup to see if the about:sync addon is able to detect a sync error and provide some logs to report in this issue.
